src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,176 @@
+        for (int i = 0; i < index; i++) { // Проходимо до потрібного вузла
+            current = current.next;
+        }
+        return current.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index out of bounds: " + index);
+        }
+        Node<T> current = head;
+        for (int i = 0; i < index; i++) { // Проходимо до потрібного вузла
+            current = current.next;
+        }
+        T oldValue = current.value;
+        current.value = value; // Змінюємо значення вузла
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index out of bounds: " + index);
+        }
+        Node<T> current = head;
+        if (index == 0) { // Видаляємо перший вузол
+            T value = head.value;
+            head = head.next;
+            if (head != null) {
+                head.prev = null;
+            } else {
+                tail = null;
+            }
+            size--;
+            return value;
+        }
+        if (index == size - 1) { // Видаляємо останній вузол
+            T value = tail.value;
+            tail = tail.prev;
+            if (tail != null) {
+                tail.next = null;
+            } else {
+                head = null;
+            }
+            size--;
+            return value;
+        }
+        for (int i = 0; i < index; i++) { // Проходимо до потрібного вузла
+            current = current.next;
+        }
+        T value = current.value;
+        current.prev.next = current.next;
+        current.next.prev = current.prev;
+        size--;
+        return value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = head;
+        while (current != null) {
+            if ((object == null && current.value == null) ||
+                    (object != null && object.equals(current.value))) {
+                if (current == head) { // Видаляємо голову
+                    head = head.next;
+                    if (head != null) {
+                        head.prev = null;
+                    } else {
+                        tail = null;
+                    }
+                } else if (current == tail) { // Видаляємо хвіст
+                    tail = tail.prev;
+                    if (tail != null) {
+                        tail.next = null;
+                    } else {
+                        head = null;
+                    }
+                } else { // Видаляємо вузол у середині
+                    current.prev.next = current.next;
+                    current.next.prev = current.prev;
+                }
+                size--;
+                return true;
+            }
+            current = current.next;
+        }
         return false;
```

Comment:
In the `remove(T object)` method, if the object is not found in the list, the method should return false. The current implementation correctly returns false if the object is not found, but ensure that the logic for checking equality is robust, especially when dealing with null values.